### PR TITLE
Library for IP -> ASN mapping

### PR DESCRIPTION
**Pull request by aarshkshah1992**
ONLY supports IPv6 for now.

### CODE REVIEW

File: asn.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+}
+
+type asnStore struct {
+	cr cidranger.Ranger
+}
+
+// AsnForIPv6 returns the AS number for the given IPv6 address.
+// If no mapping exists for the given IP, this function will
+// return an empty ASN and a nil error.
+func (a *asnStore) AsnForIPv6(ip net.IP) (string, error) {
```
Comment (willscott) [inline asn.go:33]: string is a weird return type? Asn's are positive integers

File: generate/main.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+func writeMappingToFile(f *os.File, m map[string]string, mapName string) {
+	printf := func(s string, args ...interface{}) {
+		_, err := fmt.Fprintf(f, s, args...)
+		if err != nil {
+			panic(err)
+		}
+	}
+	printf("package %s\n\n", pkgName)
```
Comment (willscott) [inline generate/main.go:46]: consider `template`, which may make this easier to read:
e.g. https://github.com/ipfs/go-datastore/blob/master/fuzz/cmd/generate/main.go#L60

File: README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Table of Contents
+
+- [Install](#install)
+- [Usage](#usage)
+- [Documentation](#documentation)
+- [Contribute](#contribute)
+- [License](#license)
+
+## Install
```
Comment (willscott) [inline README.md:18]: include a section on how to rebuild the DB

**Review by willscott — COMMENTED** [review]
* don't include the tsv in the repo
* ideally set up so `go generate` rebuilds the generated file
* would be great if the rebuilder can re-download a new version of the database, so that `go generate` is the only command that needs to be run to update the lib.

File: asn.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	// longest prefix match
+	n := ns[len(ns)-1].(*networkWithAsn)
+	return n.asn, nil
+}
+
+// NewAsnStore returns a `asnStore` that can be queried for the Autonomous System Numbers
+// for a given IP address or a multiaddress which contains an IP address.
+func NewAsnStore() (*asnStore, error) {
```
Comment (Stebalien) [inline asn.go]: Is there any way to pre-generate this? This only adds 2MiB of data (not nice but not terrible), but this function takes _way_ too long to run.

**Comment by jtacoma** [comment]
Note this would also make the master branch non-empty, solve #2, and allow a simple `go get package/that/depends/on/this` to get a non-empty go-libp2p-asn-util.

**Pull request merged**